### PR TITLE
Update Nix to 22.05

### PR DESCRIPTION
**Pull request by wd15**
A [change was required](https://github.com/materialsinnovation/pymks/compare/master...wd15:fix-nix?expand=1#diff-79aedac665912578faffe439d6f600b8a12a5c0e140b05f41f95f6c897912f43L753) due to a change in the Sfepy API. Using a custom Sfepy version (not Nix version as that's broken in 22.05 (but fixed on master)).

Other things:

 - Other code changes are for the linters
 - Switch to use mamba instead of conda for building from environment.yml
 - Use GitHub Actions for tests

### CODE REVIEW

**Comment by wd15** [comment]
The actions is running here, https://github.com/wd15/pymks/actions

It's not showing up in the boxes for some reason. Probably will work when this is merged.

**Comment by wd15** [comment]
@beyucel : ready for your review now

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,7 +14,8 @@ jobs:
       - run:
           name: Create env and install dependencies
           command: |
-            conda env create -f environment.yml
+            conda install mamba -n base -c conda-forge
+            mamba env create -f environment.yml
```
Comment (beyucel) [inline .circleci/config.yml:18]: Is mamba as stable as mamba?  I heard that it is faster but don't knot much about it. Snakemake is using mamba as well.

**Comment by beyucel** [comment]
@wd15  I ran all the tests (code and notebooks). Only the graph-descriptors notebook failed. I think failure happened due to this error/warning `distributed.nanny - WARNING - Restarting worker` [on cells 11 and 12]. 

It computed the descriptors after the warning but got stuck at line 23 `pipeline.fit`. (I only ran that line 23 for 10 min)

I know this update is about fixing the sfepy but I wanted to let you know.

**Review by beyucel — APPROVED** [review]

**Comment by wd15** [comment]
> @wd15 I ran all the tests (code and notebooks). Only the graph-descriptors notebook failed. I think failure happened due to this error/warning `distributed.nanny - WARNING - Restarting worker` [on cells 11 and 12].
> 
> It computed the descriptors after the warning but got stuck at line 23 `pipeline.fit`. (I only ran that line 23 for 10 min)
> 
> I know this update is about fixing the sfepy but I wanted to let you know.

Probably memory issues?

File: .circleci/config.yml (hunk 1 above)
Comment (wd15) [reply .circleci/config.yml:18]: I think it's quite widely used. It doesn't hang all the time like Conda. It still uses conda-forge so the packages are the same.

**Pull request merged**